repositories.yaml

```diff
@@ -177,13 +177,11 @@ libraries:
       - netwerk/metrics.yaml
       - netwerk/protocol/http/metrics.yaml
       - toolkit/components/cookiebanners/metrics.yaml
-      - toolkit/components/crashes/metrics.yaml
```

Comment:
I'm not 100% sure what happens with apps that rely on this dependency regarding the table-generation, so hearing from @relud would make me feel a lot more comfortable about how this affects that and what problems we might encounter.

My guess is that it _should_ work, since Desktop is still declaring this metrics.yaml file, just in the application definition instead of in this dependency, and the Android apps that consume `gecko` should still have these metrics definitions from the metrics.yaml file in libcrash. So it should amount to just moving things around, but I don't know if the data-infrastructure generation that happens is affected by this or not.

Reply:
As I understand it, due to push mode the results of historical scans of metrics|ping|tag files that are no longer referenced in repositories.yaml will remain in place, so removing these from gecko should be fine.

But, adding these files to an application that depends on gecko would cause duplicated metric identifiers.

After doing some work to fix duplicate metric checks, I was able to produce the following email message in testing:
```
New Email
    From: telemetry-alerts@mozilla.com
    To: crash-reporting-wg@mozilla.org,chutten@mozilla.com,stability@mozilla.org,glean-team@mozilla.com
    Subject: Glean: Duplicate metric identifiers detected
    Body:
Glean has detected duplicated metric identifiers coming from the product 'firefox-desktop'.

- 'crash.process_type' defined more than once in firefox-desktop, gecko
- 'crash.startup' defined more than once in firefox-desktop, gecko
- 'crash.time' defined more than once in firefox-desktop, gecko
- 'crash.uptime' defined more than once in firefox-desktop, gecko

What to do about this:

1. File a bug to track your investigation. You can just copy this email into the bug Description to get you started.
2. Reply-All to this email to let the list know that you are investigating. Include the bug number so we can help out.
3. Rename the most recently added metric to be more specific. See [1]
4. Make sure a Glean team member reviews any patches. Care needs to be taken that the resolution of this problem is schema-compatible.

If you have any problems, please ask for help on the #glean Slack channel. We'll give you a hand.

What this is:

We have a system called probe-scraper [2] that scrapes the metric information from all Mozilla products using the Glean SDK. All the scraped data is available on the probeinfo service [3]. The scraped definition is used to build things such as the probe-dictionary [4] and other data tools. It detected that one metric that was recently added has an identifier collision with some metric that already existed in the application namespace. So it sent this email out, encouraging you to fix the problem.

What happens if you don't fix this:

The metrics will compete to send their data in pings, making the data unreliable at best.

You can do this!

Your Friendly, Neighborhood Glean Team

[1] - https://mozilla.github.io/glean/book/user/adding-new-metrics.html#naming-things
[2] - https://github.com/mozilla/probe-scraper
[3] - https://probeinfo.telemetry.mozilla.org/
[4] - https://telemetry.mozilla.org/probe-dictionary/
```

Reply:
@relud Does it matter that this is firefox_desktop and gecko which don't use push-mode?

Reply:
they use update mode, which is essentially push mode with daily batches, so they would still be impacted.